src/libbluechi/common/cfg.c

```diff
@@ -326,7 +326,7 @@ int cfg_s_set_value(
                         return -ENOMEM;
                 }
         }
-        struct config_option *replaced = hashmap_set(
+        struct config_option *replaced = (struct config_option *) hashmap_set(
```

Comment:
I verified that your changes are indeed completely fixing #459 - nicely done! 
Only the `hashmap.c` submodule would be needed to change to v0.7.0. 

> Do you think it should be reverted?

For this PR, yes please revert it. Since this is something unrelated to the dbus return message and a library version needs to change, it should go into a dedicated PR. 
So, I'd ask you to:
- move the hashmap.c changes to a new branch and create another PR
- revert the hashmap.c changes in this PR

Would you be up for that? If you need any support on the way, just reach out, of course :)

Reply:
The changes pertaining to hashmap.c requires me to update the version to v0.7.0 and proceed with this: "move the hashmap.c changes to a new branch and create another PR", correct? @engelmi

Reply:
Yes, exactly (and update to version v0.7.0 in the new branch, of course). Thanks! @Iiqbal2000